continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfiler.java

```diff
@@ -15,20 +15,24 @@
  */
 package com.github.krzysztofslusarski.asyncprofiler;
 
-import lombok.extern.slf4j.Slf4j;
-import one.profiler.AsyncProfiler;
-import org.springframework.beans.factory.DisposableBean;
-import org.springframework.util.StringUtils;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import one.profiler.AsyncProfiler;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.util.StringUtils;
 
 @Slf4j
 public class ContinuousAsyncProfiler implements DisposableBean {
-
-    private final List<Thread> threads = new ArrayList<>();
+    private final List<ScheduledFuture<?>> scheduledFutures = new ArrayList<>();
+    private final ScheduledExecutorService executorService = new ScheduledThreadPoolExecutor(4);
```

Comment:
How do you feel to lower number of threads in pool? There is 4 job one runs every 30s, another every 10m, then 1h and 1d. Even if one of them waits a bit longer it shouldn't interfere on profiling. Am I wrong?

Reply:
done